src/components-examples/material/tree/tree-loadmore/tree-loadmore-example.ts

```diff
@@ -132,57 +158,58 @@ export class TreeLoadmoreExample {
 
-  transformer = (node: LoadmoreNode, level: number) => {
-    const existingNode = this.nodeMap.get(node.item);
+  transformer = (node: NestedNode, level: number) => {
+    const existingNode = this.nodeMap.get(node.name);
 
     if (existingNode) {
       return existingNode;
     }
 
-    const newNode = new LoadmoreFlatNode(
-      node.item,
-      level,
-      node.hasChildren,
-      node.loadMoreParentItem,
-    );
-    this.nodeMap.set(node.item, newNode);
+    const newNode = new FlatNode(node.name, level, node.hasChildren, node.parent, node.isLoadMore);
+    this.nodeMap.set(node.name, newNode);
     return newNode;
   };
 
-  getLevel = (node: LoadmoreFlatNode) => node.level;
+  getLevel = (node: FlatNode) => node.level;
 
-  isExpandable = (node: LoadmoreFlatNode) => node.expandable;
+  isExpandable = (node: FlatNode) => node.expandable;
 
-  hasChild = (_: number, _nodeData: LoadmoreFlatNode) => _nodeData.expandable;
+  hasChild = (_: number, node: FlatNode) => node.expandable;
 
-  isLoadMore = (_: number, _nodeData: LoadmoreFlatNode) => _nodeData.item === LOAD_MORE;
+  isLoadMore = (_: number, node: FlatNode) => node.isLoadMore;
 
-  private loadMoreData(node: LoadmoreFlatNode) {
-    // TODO: set focus to appropriate location
-    if (node.loadMoreParentItem) {
-      this._database.loadMore(node.loadMoreParentItem);
-    }
+  loadChildren(node: FlatNode) {
+    this._database.loadChildren(node.name, true);
   }
 
-  /** Load more nodes from data source */
-  loadMoreOnClick(event: MouseEvent, node: LoadmoreFlatNode) {
-    this.loadMoreData(node);
+  /** Load more nodes when clicking on "Load more" node. */
+  loadOnClick(event: MouseEvent, node: FlatNode) {
+    this.loadSiblings(event.target as HTMLElement, node);
   }
 
-  loadMoreOnEnterOrSpace(event: KeyboardEvent, node: LoadmoreFlatNode) {
+  /** Load more nodes on keyboardpress when focused on "Load more" node */
+  loadOnKeypress(event: KeyboardEvent, node: FlatNode) {
     if (event.keyCode === ENTER || event.keyCode === SPACE) {
-      // TODO: set focus to an appropriate location
-      this.loadMoreData(node);
-
-      // Prevent default behavior so that the tree node doesn't handle the keypress instead of this
-      // button.
-      event.preventDefault();
+      this.loadSiblings(event.target as HTMLElement, node);
     }
   }
 
-  loadChildren(node: LoadmoreFlatNode) {
-    this._database.loadMore(node.item, true);
+  private loadSiblings(nodeElement: HTMLElement, node: FlatNode) {
+    if (node.parent) {
+      // Store a reference to the sibling of the "Load More" node before it is removed from the DOM
+      const previousSibling = nodeElement.previousElementSibling;
+
+      // Synchronously load data.
+      this._database.loadChildren(node.parent);
+
+      const focusDesination = previousSibling?.nextElementSibling || previousSibling;
+
+      if (focusDesination) {
+        // Restore focus.
+        (focusDesination as HTMLElement).focus();
```

Comment:
the one thing I'm concerned about here is whether or not the tabindex updates correctly.

Reply:
I added a unit test to confirm that the tabindex updates.

Reply:
nvm, that's something else. Let me double check that manually right now

Reply:
I tested the Load More example, it it correctly updates tabindex when clicking "Load More". It focuses the first new item that appears and sets it's tabindex to 0. All other tree nodes have tabindex="-1".

There's also a unit test to verify that CdkTreeNode correctly responds to focus event. Calling `.focus()` will trigger native focus event.